### PR TITLE
#4420: eager EnqueueHostToDeviceTransfer impl

### PR DESCRIPTION
**Pull request by TT-billteng**

### CODE REVIEW

File: tt_eager/queue/queue.cpp [hunk 1]
```diff
@@ -1,29 +1,42 @@
...
-
+   void QueueSynchronize(CommandQueue& q){
+        Finish(q);
+    }
 
-    void EnqueueOperation(Queue&q, operation::DeviceOperation& devop, const std::vector<Tensor>& input_tensors, const std::vector<Tensor>& output_tensors){}
-    void EnqueueOperation(Queue&q, operation::DeviceOperation& devop, const std::vector<Tensor>& input_tensors, const std::vector<std::optional<Tensor>>& optional_input_tensors, const std::vector<Tensor>& output_tensors){}
+    void QueueRecordEvent(CommandQueue& q, Event&e){}
+    void QueueWaitForEvent(CommandQueue& q, Event&e){}
+    void EventSynchronize(Event&e){}
```
Comment (eyonland) [inline tt_eager/queue/queue.cpp:32]: @TT-billteng, are there plans to get this functionality implemented?

Comment (TT-billteng) [reply tt_eager/queue/queue.cpp:32]: yes in the future, they wanted stubs for now

**Comment by TT-billteng** [comment]
@razorback3

**Review by arakhmati — APPROVED** [review]

**Comment by TT-billteng** [comment]
getting failures @arakhmati https://github.com/tenstorrent-metal/tt-metal/actions/runs/7940176867/job/21681122239

**Pull request merged**